src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol_ImplementationChecks.cs

```diff
@@ -1259,6 +1259,21 @@ private bool IsValidOverrideReturnType(Symbol overridingSymbol, TypeWithAnnotati
                 }
             }
 
+            TypeWithAnnotations getNotNullIfNotNullOutputType(TypeWithAnnotations outputType, ImmutableHashSet<string> notNullIfParameterNotNull)
+            {
+                for (var i = 0; i < overriddenParameters.Length; i++)
```

Comment:
Can probably wrap this in an `if (!notNullIfParameterNotNull.IsEmpty)` so that we only enumerate parameters if necessary.